src/application-insights/azext_applicationinsights/tests/latest/test_applicationinsights_commands.py

```diff
@@ -0,0 +1,53 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=line-too-long
+from azure.cli.testsdk import ScenarioTest
+
+
+class ApplicationInsightsDataClientTests(ScenarioTest):
+    """Test class for Application Insights data client."""
+    def test_query_execute(self):
+        """Tests data plane query capabilities for Application Insights."""
+        self.cmd('az monitor app-insights query --apps 578f0e27-12e9-4631-bc02-50b965da2633 f4963800-c77d-40a3-8b0b-448678904c33 --analytics-query "availabilityResults | distinct name | order by name asc"', checks=[
+            self.check('tables[0].rows[-1][0]', 'microsoft'),
+            self.check('tables[0].rows[-2][0]', 'google')
+        ])
+        query_guid = self.cmd('az monitor app-insights query --app 578f0e27-12e9-4631-bc02-50b965da2633 --analytics-query "requests | getschema"').get_output_in_json()
+        query_name_rg = self.cmd('az monitor app-insights query --apps ace-test -g ace-test --analytics-query "requests | getschema"').get_output_in_json()
+        query_azure_id = self.cmd('az monitor app-insights query --analytics-query "requests | getschema" --ids /subscriptions/b98981de-4152-480f-a515-59b099299283/resourceGroups/ace-test/providers/microsoft.insights/components/ace-test').get_output_in_json()
+        assert query_guid == query_name_rg
+        assert query_name_rg == query_azure_id
+        assert len(query_guid['tables'][0]['rows']) == 37
+        assert isinstance(query_guid['tables'][0]['rows'][0][1], (int, float, complex))
+
+    def test_metrics_show(self):
+        result = self.cmd('az monitor app-insights metrics show --app /subscriptions/b98981de-4152-480f-a515-59b099299283/resourceGroups/ace-test/providers/microsoft.insights/components/ace-test -m availabilityResults/count').get_output_in_json()
```

Comment:
This will fail the CI as we scrub the subscription id for the recordings. 
Please make this subscription agnostic.

Reply:
Makes sense. I saw the docs on preparers. Is that the route I should go/would I need to write an application preparer? It seems there were only a handful of preparers other applications were using.

Reply:
I don't think so. It seems like your scenarios require app-insights applications, which won't be visible when not logged in with your credentials on your subscriptions.

Usually, for most test scenarios, the tests themselves would create the resources needed. However, since the mgmt-plane commands won't be available yet to create those resources, this won't be possible.

I'd recommend referring to the applications by GUID for now, or, after recording, changing the subscriptions in those arm-ids, to be zeros in your test.

Reply:
@harshdeepgupta for recorded tests, we should strip sensitive information manually for now

Reply:
I've scrubbed these manually, and left the application GUIDs.